app/src/main/java/ch/epfl/favo/chat/ChatPage.java

```diff
@@ -124,13 +130,39 @@ public void onItemRangeInserted(int positionStart, int itemCount) {
   }
 
   private void onSendClick() {
+    String favorId = currentFavor.getId();
+    String requesterNotifId = currentFavor.getRequesterNotifId();
+    String responderUserId = DependencyFactory.getCurrentFirebaseUser().getUid();
+
+//    boolean isFirstMsg = false;
+//    CollectionReference chat = FirebaseFirestore.getInstance().collection("chat");
+//    Query query = chat.whereEqualTo(FAVOR_ID, favorId)
+//            .whereEqualTo(NOTIF_ID, requesterNotifId)
+//            .whereEqualTo(USER_ID, responderUserId);
+//
+//    query.get().addOnCompleteListener(task -> {
+//      if (task.isSuccessful()) {
+//        for (QueryDocumentSnapshot document : task.getResult()) {
+//          Log.d(TAG, document.getId() + " => " + document.getData());
+//        }
+//      } else {
+//        Log.d(TAG, "Error getting documents: ", task.getException());
+//      }
+//    });
+
+    // If the empty message view is visible, then this is the first message.
+    String isFirstMsg = String.valueOf(view.findViewById(R.id.emptyTextView).getVisibility() == View.VISIBLE);
```

Comment:
What if the first message is from the requester? I guess he would receive a notification for its own message if I checked correctly and that should not happen.